### PR TITLE
age-plugin-yubikey: init at 0.2.0

### PR DESCRIPTION
**Pull request by vtuan10**
###### Motivation for this change
I want to use my yubikey with (r)age.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/age-plugin-yubikey/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, pcsclite
+, PCSC
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "age-plugin-yubikey";
+  version = "unstable-2021-12-24";
```
Comment (veehaitch) [inline pkgs/tools/security/age-plugin-yubikey/default.nix:12]: Maybe just use `v0.2.0` from https://github.com/str4d/age-plugin-yubikey/commit/51910edfab4006a068864602469ff7db3766bfbe?

**Review by veehaitch — COMMENTED** [review]
Works fine with a YubiKey 5C Nano on `aarch64-darwin`.

Created a new key through the TUI of `age-plugin-yubikey`, encrypted a secret with the resulting `age1yubikey*` recipient, and decrypted it with the corresponding identity using `age-plugin-yubikey`.

File: pkgs/tools/security/age-plugin-yubikey/default.nix (hunk 1 above)
Comment (vtuan10) [reply pkgs/tools/security/age-plugin-yubikey/default.nix:12]: Unfortunately there is no release or tag `v0.2.0` 😅 
I'll create a issue on their side

Comment (veehaitch) [reply pkgs/tools/security/age-plugin-yubikey/default.nix:12]: Sure, but just use the commit as a `rev` and change `version` to `0.2.0`. Maybe add a comment that there's no tag for this release.

File: pkgs/tools/security/age-plugin-yubikey/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "age-plugin-yubikey";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "str4d";
+    repo = pname;
+    rev = "51910edfab4006a068864602469ff7db3766bfbe"; # no tag for this release
+    sha256 = "sha256-KdxghSvkIsxqADzIZL8aSw8azYvcPF0mpw78uEWG2rs=";
```
Comment (veehaitch) [inline pkgs/tools/security/age-plugin-yubikey/default.nix:18]:
```suggestion
    hash = "sha256-mMqvBlGFdwe5BaC0bXZg/27BGNmFTTYbLUHWUciqxQ0=";
```

File: pkgs/tools/security/age-plugin-yubikey/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  cargoSha256 = "sha256-vtHz3ILzGMUinmG6RlicaUGZCC0WIgLA+PfsuRlD/2c=";
```
Comment (veehaitch) [inline pkgs/tools/security/age-plugin-yubikey/default.nix:21]:
```suggestion
  cargoSha256 = "sha256-OCbVLSmGx51pJ/EPgPfOyVrYWdloNEbexDV1zMsmEJc=";
```

**Review by veehaitch — CHANGES_REQUESTED** [review]
You missed to update the hashes 🙂

**Comment by vtuan10** [comment]
> You missed to update the hashes 🙂

My bad, fixed :)

**Review by veehaitch — APPROVED** [review]

**Review by marsam — APPROVED** [review]

**Pull request merged**